### ISSUE TITLE
My understanding in test_nn_model in main.py

### ISSUE THREAD

**Issue by nxznm** [AbductiveLearning/ABL-HED#3]
In `test_nn_model()` in `main.py`, it trys to use all maps to find the best map to get the best acc. In the beginning, I was a little confused about this. But now I can explain. Let me show my opinion.

First, I want to say that the program here doesn't use any true label information to classify the image into 4 categories. In the pretrain stage, it use the `autoencoder_net`, and choose its front net structure and parameters as `base_model`. 

 Then it goes into abduce stage, it use `base_model` to translate the equations, and use prolog module to abduce. For example, prolog module may says it is not category one, it is category two. And `base_model` use this information to retrain again. 

Finally `base_model` is well enough that gives it a image, it can say it belongs to which category. The category concept which the `base_model` has isn't consistent with our human category concept(or the real category).

So, the experiment trys to traverse each map, and use the best acc as the neural network perception acc. **And this actually intimates that the map which comes to the best acc reflects the relationship between `base_model` category and the real category.** This method has its retionality. Although this inconsistence between `base_model` category and real world category looks a little strange. 

This is my problem and my opinion when I read the code. I hope it can help other people to read this project.
